bdk_chain/src/keychain.rs

```diff
@@ -20,13 +20,79 @@ mod keychain_txout_index;
 #[cfg(feature = "miniscript")]
 pub use keychain_txout_index::*;
 
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Deserialize, serde::Serialize),
+    serde(
+        crate = "serde_crate",
+        bound(
+            deserialize = "K: Ord + serde::Deserialize<'de>",
+            serialize = "K: Ord + serde::Serialize"
+        )
+    )
+)]
+#[must_use]
+pub struct DerivationAdditions<K>(BTreeMap<K, u32>);
```

Comment:
Won't this not necessarily be monotone when you add the trim methods in #104. Maybe just call it a changeset too.

Actually what's wrong with just a `BTreeMap` here? I don't see the newtype doing much here.

Reply:
> Actually what's wrong with just a BTreeMap here? I don't see the newtype doing much here.

I wanted to have the `append` logic just like the rest of our changeset/additions. I feel like it would be an extremely useful API to have.

Reply:
`BTreeMap` has `.append` though.

Reply:
Not the type of "append" we need though. We need to keep the highest value, not only set the ones that are missing.